src/test/java/org/beanpod/switchboard/controller/DeviceControllerTest.java

```diff
@@ -52,6 +53,19 @@ void setup() {
     MockitoAnnotations.initMocks(this);
   }
 
+  @Test
+  final void testUploadJson() {
+
+    when(deviceService.findDevice(any())).thenReturn(Optional.of(deviceDTO));
+    when(deviceMapper.toDeviceDto((DeviceModel) any())).thenReturn(deviceDTO);
+    when(deviceService.save(any())).thenReturn(deviceDTO);
+    when(deviceMapper.toDeviceModel(any())).thenReturn(deviceModel);
+
+    String s1 =
+        deviceController.uploadConfiguration(DeviceFixture.SERIAL_NUMBER, validJsonFile).getBody();
+    assertEquals("Configurations uploaded.", s1);
```

Comment:
Should assert somewhere that the device being saved in `deviceService` actually has the correct `configurationInstance`

Reply:
how would you verify that?

Reply:
```
DeviceDto expectedSavedDevice = DeviceFixture.getDeviceDto();
expectedSavedDevice.setConfigurationInstance(validJsonFile.getBytes());
verify(deviceService).save(expectedSavedDevice);
```

Reply:
Fixed